pkg/controller/networkpolicy/store/networkpolicy.go

```diff
@@ -122,7 +121,7 @@ func NetworkPolicyKeyFunc(obj interface{}) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("object is not *types.NetworkPolicy: %v", obj)
 	}
-	return k8s.NamespacedName(policy.Namespace, policy.Name), nil
+	return string(policy.UID), nil
```

Comment:
Let's discuss this in the other thread first? I feel this is risky and inconsistent with all other API in Antrea and K8s.

Reply:
If we want to revert this we need to revert keyFunc change in network_controller.go as well. Also we will not be able to differentiate anp and k8s np of the same name in a namespace... Verified that this does cause unexpected behaviors

Reply:
does or does not cause unexpected behavior ?

Reply:
I mean w/o changing the keyFunc, verified there are unexpected behaviors (ANP add is considered NP update etc.)

Reply:
Done reverting the change and create issue #1173